llm_on_ray/inference/deepspeed_predictor.py

```diff
@@ -244,20 +244,34 @@ def streaming_generate(self, prompt, streamer, **config):
         for worker in self.prediction_workers[1:]:
             worker.streaming_generate.remote(inputs_ref, self._create_dummy_streamer(), **config)
 
-    def generate(self, prompt, **config):
-        input_ids, input_length = self.tokenize_inputs(prompt)
+    def generate(
```

Comment:
Can we simplify this method by always accepting prompts as List[str] like below? We can write another method that accepts a prompt: str and returns `self.generate([prompt], **config)[0]`

```suggestion
    def generate(
        self, prompts: List[str], **config
    ) -> List[GenerateResult]:
        input_ids, input_length = self.tokenize_inputs(prompts)
        inputs_ref = ray.put(input_ids)
        gen_tokens = ray.get(
            [worker.generate.remote(inputs_ref, **config) for worker in self.prediction_workers]
        )[0]

        decode_result = self.tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)

        return [
            GenerateResult(
                text=decode_result[i],
                input_length=input_length,
                generate_length=gen_tokens.size()[1] - input_length,
            )
            for i in range(len(prompts))
        ]
```

Reply:
thanks, I will add to the refactor plan to always use list as input.